.github/actions/argus-builder/build-prep/src/main.ts

```diff
@@ -0,0 +1,259 @@
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import { minimatch } from 'minimatch';
+// eslint-disable-next-line import/no-relative-packages
+import { findChangedFiles } from '../../../find-changed-files/src/findChangedFiles';
+// eslint-disable-next-line import/no-relative-packages
+import { validateJsonSchema } from '../../../validate-json-schema/src/validateJsonSchema';
+// eslint-disable-next-line import/no-relative-packages
+import { getCommaDelimitedArrayInput, ProcessedImage } from '../../common';
+
+// eslint-disable-next-line import/no-relative-packages
+export { ProcessedImage } from '../../common';
+
+type Inputs = {
+  githubToken: string
+  images: Record<string, any>
+  pathFilters: string[]
+  branchesInclude: string[]
+  branchesIgnore: string[]
+  manifestTriggerLabels: string[]
+};
+
+export function getInputs(): Inputs {
+  return {
+    githubToken: core.getInput('github_token', { required: true }),
+    images: JSON.parse(core.getInput('images', { required: true })),
+    pathFilters: getCommaDelimitedArrayInput('path_filters', { required: true }),
+    branchesInclude: getCommaDelimitedArrayInput('branches_include', { required: true }),
+    branchesIgnore: getCommaDelimitedArrayInput('branches_ignore', { required: false }),
+    manifestTriggerLabels: getCommaDelimitedArrayInput('manifest_trigger_labels', { required: true }),
+  };
+}
+
+const imagesInputSchema = {
+  type: 'object',
+  additionalProperties: {
+    type: 'object',
+    additionalProperties: false,
+    properties: {
+      context: { type: 'string' },
+      dockerfile: { type: 'string' },
+      platform: { type: 'string' },
+      build_args: {
+        type: 'array',
+        items: { type: 'string' },
+      },
+      secret_files: {
+        type: 'array',
+        items: { type: 'string' },
+      },
+      argus_root: { type: 'string' },
+      path_filters: {
+        type: 'array',
+        items: {
+          oneOf: [
+            { type: 'string' },
+            { type: 'array', items: { type: 'string' } },
+          ],
+        },
+      },
+      branches_include: { type: 'array', items: { type: 'string' } },
+      branches_ignore: { type: 'array', items: { type: 'string' } },
+    },
+    required: ['context', 'dockerfile'],
+  },
+};
+
+type ImageInput = {
+  context: string
+  dockerfile: string
+  platform?: string
+  build_args?: string[]
+  secret_files?: string[]
+  argus_root?: string
+  path_filters?: string[] | string[][]
+  branches_include?: string[]
+  branches_ignore?: string[]
+};
+
+if (process.env.NODE_ENV !== 'test') {
+  main();
+}
+export async function main() {
+  const inputs = getInputs();
+  core.info(`Received input: ${JSON.stringify(inputs, null, 2)}`);
+
+  core.info('Validating images input...');
+  validateJsonSchema(inputs.images, imagesInputSchema);
```

Comment:
I'm not familiar with Ajv - inputs.images is an array right? imagesInputSchema says the root is of type object  but shouldn't it be an array?

Reply:
`inputs.images` in an object, the `processImagesInput` function converts it to an array which is written to the output and that output is what the build step uses to set up the build matrix